### DISCUSSION TITLE
Field override withProperty requires value while grafana allows only id

### DISCUSSION THREAD

**Discussion by DieBauer** [grafana/grafonnet#231] [Q&A]
I'm trying to recreate a panel with two y-axes in grafonnet, where I need to override a `max` setting. 

In grafonnet I can't express an empty field, it requires a value, and the behavior at runtime is different. 

this doesn't compile: RUNTIME ERROR: Missing argument: value
```jsonnet
fieldOverride.byName.withProperty(
          'max',
        )
```

I need the 'auto' for it to work like I want it. However adding 'auto' to jsonnet as value doesn't work as expected. 
![Image](https://github.com/user-attachments/assets/57b63aca-af8d-42f4-8566-bc026b865fd1)

This is the Panel JSON with the expected lay-out, using 'auto'. 
```json
    "overrides": [
      {
        "matcher": {
...
        },
        "properties": [
          {
            "id": "custom.axisPlacement",
            "value": "right"
          },
          {
            "id": "custom.axisLabel",
            "value": "Requests per second"
          },
          {
            "id": "max"
          },
          {
            "id": "custom.axisSoftMin",
            "value": "0"
          }
        ]
      }
    ]
```

How can I workaround this? Or should the 'withProperty' be changed to take defaults into account depending on the key. 

/edit 
in grafana the override
```json
          {
            "id": "max"
          }
```
is different from
```json
          {
            "id": "max",
            "value": "auto"
          }
```

**Comment by Duologic** [comment]
Would this example help?

```jsonet
timeSeries.standardOptions.withOverrides([
  timeSeries.standardOptions.override.byName.new('<name from legend>')
  + timeSeries.standardOptions.override.byName.withPropertiesFromOptions(
    timeSeries.fieldConfig.defaults.custom.withAxisPlacement('right')
  ),
])
```
